google-cloud-spanner/src/main/java/com/google/cloud/spanner/Value.java

```diff
@@ -1909,6 +2069,23 @@ public BigDecimal getNumeric() {
       return valueAsBigDecimal;
     }
 
+    @Override
+    public float getFloat32() {
+      checkNotNull();
+      if (floatConversionError != null) {
+        throw floatConversionError;
+      }
+      if (valueAsFloat == null) {
+        try {
+          valueAsFloat = Float.parseFloat(value);
+        } catch (NumberFormatException e) {
+          floatConversionError = e;
+          throw e;
+        }
+      }
+      return valueAsFloat;
+    }
+
```

Comment:
I think we can (and should) remove this. There's no specific reason that we would need to get a `PgNumeric` as a float32. The reason that the `getFloat64()` method is here, is that `PgNumeric` is a bit different than the standard `Numeric` in GoogleSQL Spanner; `PgNumeric` also supports NaN, which cannot be represented as a BigDecimal.

```suggestion
```

Reply:
Done